js/map.js

```diff
@@ -0,0 +1,47 @@
+var marker;
+var infowindow;
+
+function myMap() {
+    var mapCanvas = document.getElementById("map");
+    var myCenter = new google.maps.LatLng(32.7281, -117.1833);
+    var mapOptions = {
+        center: myCenter, zoom: 10,
+        mapTypeId: google.maps.MapTypeId.ROADMAP
+    };
+    var map = new google.maps.Map(mapCanvas, mapOptions);
+    google.maps.event.addListener(map, 'click', function (event) {
+        placeMarker(map, event.latLng);
+    });
+}
+
+function placeMarker(map, location) {
+    if (!marker || !marker.setPosition) {
+        marker = new google.maps.Marker({
+            position: location,
+            map: map,
+        });
+    } else {
+        marker.setPosition(location);
+    }
+    if (!!infowindow && !!infowindow.close) {
+        infowindow.close();
+    }
+    infowindow = new google.maps.InfoWindow({
+        content: 'Latitude: ' + Math.round(location.lat() * 10000) / 10000 + '<br>Longitude: ' + Math.round(location.lng() * 10000) / 10000
+
+    });
+    document.getElementById('latitude').value = Math.round(location.lat() * 10000) / 10000;
+    document.getElementById('longtitude').value = Math.round(location.lng() * 10000) / 10000;
+    infowindow.open(map, marker);
+}
+
+function renderStaticMap() {
+    document.getElementById("staticmap").innerHTML +=

```

Comment:
innerHTML is a dangerous property. It could potentially allow a user to break a site if it takes in user-generated data. Double check that a malicious user can't inject hazardous code into our site when they use this.

Reply:
I totally have no idea whether innerHTML is dangerous and my coding skill is at a low level, also we are running out of time now. if you have an alternative or other solutions for this problem, please let me know.

Reply:
Here's a good article to read when you have time: https://betterprogramming.pub/dom-manipulation-the-dangers-of-innerhtml-602f4119d905

It's fine for now, and I think there isn't much security risk since we should be the only ones setting any values into the staticmap element, but it is something important to consider when you're dealing with user-generated values.